docs/content/guides/dagster/pythonic-config.mdx

```diff
@@ -0,0 +1,225 @@
+
+It's often useful to provide user-chosen values to Dagster jobs or software-defined assets at runtime. For example, you might want to choose what dataset an op runs against, or provide a connection URL for a database resource. Dagster exposes this functionality through a configuration API.
+
+## Usinig Pythonic configuration
+
+### Defining Pythonic configuration on ops and assets
+
+Configurable parameters accepted by an op or asset are specified by defining a config model subclass of <PyObject object="Config"/> and a `config` parameter to the corresponding op or asset function. Under the hood, these config models utilize [Pydantic](https://docs.pydantic.dev/), a popular Python library for data validation and serialization.
+
+During execution, the specified config is accessed within the body of the op or asset using the `config` parameter.
+
+<TabGroup persistentKey="assetsorops">
+<TabItem name="Using software-defined-assets">
+
+#### Using software-defined assets
+
+Here, we define a subclass of <PyObject object="Config"/> holding a single string value representing the name of a user. We can access the config through the `config` parameter in the asset body.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_basic_asset_config endbefore=end_basic_asset_config
+class MyAssetConfig(Config):
+    person_name: str
+
+
+@asset
+def greeting(config: MyAssetConfig) -> str:
+    return f"hello {config.person_name}"
+```
+
+</TabItem>
+<TabItem name="Using ops and jobs">
+
+#### Using ops
+
+Here, we define a subclass of <PyObject object="Config"/> holding a single string value representing the name of a user. We can access the config through the `config` parameter in the op body.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_basic_op_config endbefore=end_basic_op_config
+class MyOpConfig(Config):
+    person_name: str
+
+
+@op
+def print_greeting(config: MyOpConfig):
+    print(f"hello {config.person_name}")
+```
+
+</TabItem>
+</TabGroup>
+
+---
+
+### Defining and accessing configuration for a resource
+
+Configurable parameters for a resource are defined by specifying attributes for a resource class, which subclasses <PyObject object="ConfigurableResource"/>. The below resource defines a configurable connection URL, which can be accessed in any methods defined on the resource.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_basic_resource_config endbefore=end_basic_resource_config
+class MyDatabaseResource(ConfigurableResource):
+    connection_url: str
+
+    def query(self, query: str) -> Response:
+        return get_engine(self.connection_url).execute(query)
+```
+
+For more information on using resources, see [the Pythonic resources guide](/guides/dagster/pythonic-resources).
+
+---
+
+### Specifying runtime configuration in Python
```

Comment:
Is `in Python` necessary? To me, this is all occurring in Python, so I'm not sure why this needs to be specified. I'd love more info if my perspective is wrong/incomplete though!

Reply:
Users can also specify runtime configuration in the launchpad - this is specific to specifying configuration in code. Will change to `in code` since that's more clear.

Reply:
🤦‍♀️ That's a good point - thanks for clarifying! I like `in code` here.